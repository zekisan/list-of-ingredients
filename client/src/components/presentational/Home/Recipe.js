import React from 'react';
import PropTypes from 'prop-types';
import { Container, Header, Segment, Button } from 'semantic-ui-react';

const Recipe = ({ drink }) => <span>
  {drink &&
    <Container>
      <Header as='h2'>{drink.title}</Header>
      {drink.description && <p>{drink.description}</p>}
      {drink.ingredients &&
        <Segment.Group>
          {drink.ingredients.map((ingredient, i) => <Segment key={i}>{ingredient.description}</Segment>)}
        </Segment.Group>
      }
      {drink.steps && <p>{drink.steps}</p>}
      {drink.source && <Button basic size='tiny' color='teal' href={drink.source}>Source</Button>}
    </Container>
  }
</span>;

Recipe.propTypes = {
  /** the drink data */
  drink: PropTypes.object,
};

export default Recipe;