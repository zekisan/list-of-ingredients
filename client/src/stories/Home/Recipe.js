import React from 'react';

import { storiesOf } from '@storybook/react';
import Recipe from '../../components/presentational/Home/Recipe';

import drinks from '../../mocks/drinks'

storiesOf('Recipe', module).add('to Storybook', () => <Recipe drink={drinks[0]}/>);