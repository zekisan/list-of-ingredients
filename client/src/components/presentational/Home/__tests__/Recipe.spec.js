import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Recipe from '../Recipe';

describe('Recipe', () => {
    let props;
    let mountedRecipe;
    const recipe = () => {
        if(!mountedRecipe) {
            mountedRecipe = mount(<Recipe { ...props } />);
        }
        return mountedRecipe;        
    }
    
    beforeEach(() => {
        props = { drink: undefined }
        mountedRecipe = undefined;
    });

    it('always renders a span', () => {
        const spans = recipe().find('span');
        expect(spans.length).toBeGreaterThan(0);
    });

    describe('snapshot', () => {
        it('tries snapshots', () => {
            const tree = renderer
                .create(<Recipe />)
                .toJSON();
            
            expect(tree).toMatchSnapshot();
        });
    });
});