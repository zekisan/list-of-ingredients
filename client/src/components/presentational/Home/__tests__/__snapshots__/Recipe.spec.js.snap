// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe snapshot tries snapshots 1`] = `<span />`;
